### ISSUE TITLE
tree_view_ref stays in context for opened window

### ISSUE THREAD

**Issue by satautiv** [odoo/odoo#20805]
Impacted versions: 11.0

Steps to reproduce:
(This is just an example(which does not make much sense) to show the problem).

I create a form view with one2many field where i specify a tree view with context variable tree_view_ref and don't specify form view. For example I edit stock.picking view 'stock.view_picking_form' field "move_line". I removed tree view from that view and specified tree view through context with tree_view_ref. 

![stock_picking_view](https://user-images.githubusercontent.com/15228157/32572880-c10a1a38-c4d4-11e7-96bb-c4da034f4cf2.png)

So when i click on stock move in picking form it should open that move in dialog window. And if stock.move form view has any one2many field for example "scrap_ids" - an error will occur. I added field "scrap_ids" to stock.move form view 'stock.view_move_picking_form'

![stock_move_form_view](https://user-images.githubusercontent.com/15228157/32573203-e05ea542-c4d5-11e7-8260-596267daaf04.png)

Error will not pop up in browser it could only be seen in odoo log. error:

2017-11-08 20:02:17,938 12927 INFO test11 odoo.addons.base.ir.ir_ui_view: Field "is_locked" does not exist
Error context:
View "n/a"
[view_id: n/a, xml_id: n/a, model: n/a, parent_id: n/a]
2017-11-08 20:02:17,938 12927 ERROR test11 odoo.http: Exception during JSON request handling.
Traceback (most recent call last):
  File "/home/vytautas/sandas/openerp-server-11.0/odoo/http.py", line 646, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/home/vytautas/sandas/openerp-server-11.0/odoo/http.py", line 307, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/home/vytautas/sandas/openerp-server-11.0/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/home/vytautas/sandas/openerp-server-11.0/odoo/http.py", line 683, in dispatch
    result = self._call_function(**self.params)
  File "/home/vytautas/sandas/openerp-server-11.0/odoo/http.py", line 339, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/home/vytautas/sandas/openerp-server-11.0/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/home/vytautas/sandas/openerp-server-11.0/odoo/http.py", line 332, in checked_call
    result = self.endpoint(*a, **kw)
  File "/home/vytautas/sandas/openerp-server-11.0/odoo/http.py", line 927, in __call__
    return self.method(*args, **kw)
  File "/home/vytautas/sandas/openerp-server-11.0/odoo/http.py", line 512, in response_wrap
    response = f(*args, **kw)
  File "/home/vytautas/sandas/openerp-server-11.0/addons/web/controllers/main.py", line 920, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/home/vytautas/sandas/openerp-server-11.0/addons/web/controllers/main.py", line 912, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/home/vytautas/sandas/openerp-server-11.0/odoo/api.py", line 687, in call_kw
    return call_kw_model(method, model, args, kwargs)
  File "/home/vytautas/sandas/openerp-server-11.0/odoo/api.py", line 672, in call_kw_model
    result = method(recs, *args, **kwargs)
  File "/home/vytautas/sandas/openerp-server-11.0/odoo/models.py", line 1224, in load_views
    for [v_id, v_type] in views
  File "/home/vytautas/sandas/openerp-server-11.0/odoo/models.py", line 1224, in <dictcomp>
    for [v_id, v_type] in views
  File "/home/vytautas/sandas/openerp-server-11.0/odoo/models.py", line 1310, in fields_view_get
    xarch, xfields = View.postprocess_and_fields(self._name, etree.fromstring(result['arch']), view_id)
  File "/home/vytautas/sandas/openerp-server-11.0/odoo/addons/base/ir/ir_ui_view.py", line 919, in postprocess_and_fields
    self.raise_view_error(message, view_id)
  File "/home/vytautas/sandas/openerp-server-11.0/odoo/addons/base/ir/ir_ui_view.py", line 489, in raise_view_error
    raise ValueError(message)
ValueError: Field "is_locked" does not exist

is_locked is stock.move field and is in specified tree view but it is cheked when rendering tree view for stock.scrap object.

Basically context from stock.picking view with `'tree_view_ref': 'stock.stock_move_tree_for_test'` stays and is not replaced with `context="{'tree_view_ref': 'stock.stock_scrap_tree_view'}"` when rendering stock.move dialog view and function _fields_view_get in /odoo/models.py selects wrong tree view for stock.scrap object. It selects stock.stock_move_tree_for_test view because this external ID is found in context.

It used to work in odoo 9.0 . Somehow i need to renew the context with new variables when i open one2many line.

Current behavior:

Expected behavior:

Video/Screenshot link (optional):
stock.move tree view used in this example:
![stock_move_tree_view_for_test](https://user-images.githubusercontent.com/15228157/32573889-372bab66-c4d8-11e7-9526-e22bc1537ae0.png)

**Comment by lukbas** [comment]
I am having the same issue :/

**Comment by auyeung112002** [comment]
This issue also affect version 12

**Comment by C3POdoo** [comment]
Dear @satautiv,

Thank you for your report but we are closing it due to inactivity.
We apology if we could not look at your request in time.
If your report still makes sense, don't hesitate to reopen a new one. We will try to check it as soon as possible.

_This is an automated message._
